### ISSUE TITLE
No --mount option with "podman build" nor Containerfile

### ISSUE THREAD

**Issue by fatherlinux** [containers/podman#11121]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Podman should support the --mount option with the "podman build" command and in the Containerfile using a RUN directive (similar to secrets)

**Steps to reproduce the issue:**

1. Build a Containerfile like this:
FROM registry.access.redhat.com/ubi8/ubi
RUN --mount=type=image,source=registry.fedoraproject.org/fedora:latest,destination=/mnt,rw=true
RUN yum install -y openssl

2. Option #1: Run this command: podman build .

3. Option #2: Run this command: podman build --mount=type=image,source=registry.fedoraproject.org/fedora:latest,destination=/mnt,rw=true .

**Describe the results you received:**
Option #1 will give the following error: Error: error building at STEP "RUN --mount=type=image,source=registry.fedoraproject.org/fedora:latest,destination=/mnt,rw=true ": error resolving mountpoints for container "0280ebccf2a8f4637e563dabb957e6d73e7313d548b309944dd357f6949d61ed": invalid filesystem type "image"

Option #2 will give the following error: Error: unknown flag: --mount

**Describe the results you expected:**
I would expect podman to mount the image source during a build.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:      3.2.3
API Version:  3.2.3
Go Version:   go1.16.5
Built:        Fri Jul 16 16:33:32 2021
OS/Arch:      linux/amd64

```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.21.3
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.29-2.fc34.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: '
  cpus: 1
  distribution:
    distribution: fedora
    version: "34"
  eventLogger: journald
  hostname: fedora.dc2.crunchtools.com
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.11.12-300.fc34.x86_64
  linkmode: dynamic
  memFree: 637464576
  memTotal: 2072940544
  ociRuntime:
    name: crun
    package: crun-0.18-5.fc34.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.18
      commit: 808420efe3dc2b44d6db9f1a3fac8361dde42a95
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 2071982080
  swapTotal: 2071982080
  uptime: 2h 19m 53.35s (Approximately 0.08 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 3
    paused: 0
    running: 0
    stopped: 3
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageStore:
    number: 2
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.2.3
  Built: 1626467612
  BuiltTime: Fri Jul 16 16:33:32 2021
  GitCommit: ""
  GoVersion: go1.16.5
  OsArch: linux/amd64
  Version: 3.2.3
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.2.3-1.fc34.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

No

**Additional environment details (AWS, VirtualBox, physical, etc.):**
KVM machine running Fedora 34 on top of RHEL 8.4

**Comment by rhatdan** [comment]
@ashley-cui PTAL, weird thing here is that this is going cross platform.  Currently we get --mount support from buildah,  Buildah does not currently understand a mounted image.

**Comment by ashley-cui** [comment]
I don't think either of these are implemented in buildah? `buildah bud -h` doesn't show a --mount flag, and an image as a RUN mount is supported yet; Docker doesn't have the mount=type=image option in the dockerfile spec either. 

 I'd need more clarification on what this would mean too.

I don't think RUN secrets are wired into podman yet either, only in buildah.

**Comment by rhatdan** [comment]
Yes the RUN Secrets will get vendored into Podman as soon as we vendor, which should have just happened podman 3.3

type=image is a podman specific thing.

We have podman image mount ...  Which is what Scott is thinking.

**Comment by vrothberg** [comment]
Do we really need to support `type=image` during builds?  The issue states that "Podman should support the --mount option" which it does.  It just doesn't support `type=image`.

I don't think that mounting images like that _during builds_ makes sense since this is already supported via multi-stage builds (e.g., `COPY $PATH FROM $PREVIOUS_STAGE`).

**Comment by fatherlinux** [comment]
@vrothberg  yeah, we need type=image so that we can build on UBI Micro without using multi-stage builds. The multi-stage builds are bad because they overwrite everything in UBI Micro (in the second stage).

If we could mount an image at build, we could execute the YUM/DNF/RPM commands from UBI Standard, and add files to UBI Micro.
